engine/src/main/java/org/terasology/input/cameraTarget/TargetSystem.java

```diff
@@ -44,6 +44,15 @@ public TargetSystem(BlockEntityRegistry blockRegistry, Physics physics) {
         this.physics = physics;
     }
 
+    /**
+     * Gets the position of the block that is currently targeted. If there is currently no target, this will return
+     * a null reference (isTargetAvailable() would have returned false in this case).
+     * @return the target block position in world coordinates, a vector of 3 integers.
+     */
+    public Vector3i getTargetBlockPosition() {
```

Comment:
I'll refer you to a comment I wrote on line 130 in `CharacterSystem.java`:

```
// It would seem to make more sense to just call getTargetBlock() and get the entity here, but that causes
 // a NPE up in the EntityRef.
 // Vector3i positionVector = targetSystem.getTargetBlockPosition();
 Vector3i positionVector = targetSystem.getTargetBlockPosition();
```

I'll expand that explanation here to note that replacing the  contents of the `PlayerTargetSystem.getBlockPosition()` method with:

```
Vector3i positionVector = targetSystem.getTargetBlock().
LocationComponent location = target.getComponent(LocationComponent.class);
return location.getWorldPosition();
```

causes the same NPE that is referenced above. Specifically, the NPE happens when `PlayerTargetSystem.getBlockPosition()` is called from `CharacterSystem as it was above; curiously, any calls to this method from `updateTarget()` succeed.

I traced the issue up the stack and was unable to resolve it at the time. I may report it in an issue, since it seems like this should be a valid use case of the `getTarget()` method. (I'm too lazy to go turn on my headset to get a stack trace right now, or I'd include it here :/)